lib/identity_cache/cached/primary_index.rb

```diff
@@ -60,7 +60,6 @@ def load_one_from_db(id)
       def load_multi_from_db(ids)
         return {} if ids.empty?
 
-        ids = ids.map { |id| model.connection.type_cast(id, id_column) }
```

Comment:
I tried running the tests `puts caller if ids != ids.map { |id| model.connection.type_cast(id, id_column) }` to see if we were failing to coerce the ids before building the cache keys, since that would cause more problems.  In the process, I found a couple of tests for this line, but they were just calling this private method directly with heavy mocking/stubbing, so they don't represent real usage.  I'm pushing a commit to remove those useless tests.